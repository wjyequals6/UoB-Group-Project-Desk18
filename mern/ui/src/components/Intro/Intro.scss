
@import "../../common.scss";

@keyframes shake {
    0% { transform: translate(1px, 1px) rotate(0deg); }
    10% { transform: translate(-1px, -2px) rotate(-1deg); }
    20% { transform: translate(-3px, 0px) rotate(1deg); }
    30% { transform: translate(3px, 2px) rotate(0deg); }
    40% { transform: translate(1px, -1px) rotate(1deg); }
    50% { transform: translate(-1px, 2px) rotate(-1deg); }
    60% { transform: translate(-3px, 1px) rotate(0deg); }
    70% { transform: translate(3px, 1px) rotate(-1deg); }
    80% { transform: translate(-1px, -1px) rotate(1deg); }
    90% { transform: translate(1px, 2px) rotate(0deg); }
    100% { transform: translate(1px, -2px) rotate(-1deg); }
}

@keyframes opening {
    0% { opacity: 0;}
    100% { opacity: 1; }
}

@keyframes enlarge {
    0%{transform: scale(1,1);}
    100%{transform: scale(1.05,1.05);}
}

@keyframes bounce {
	0%, 20%, 50%, 80%, 100% {transform: translateY(0);}
	40% {transform: translateY(-30px);}
	60% {transform: translateY(-15px);}
}

.Intro{
    h1{
        font-weight: 400;
        font-size: 20px;
        color: #fff;
        color: #fff;
    }
    position: fixed;
    top: 0;
    left: 0;
    box-sizing: border-box;
    width: 100%;
    height: 100%;
    background-color: $primary;
    opacity: 1;
    z-index: 200;
    transition: 1s;
    @include flexCenter;
    align-items: center;
    &.showIntro{
        opacity: 1;
        visibility: visible;
    }
    &.closeIntro{
        opacity: 0;
        visibility: hidden;
    }

    .btnText{
        user-select: none;
        cursor: pointer;
        transition: 1s;
        margin: 0 auto;
        @include flexCenter;
        // text-decoration: underline;
        font-weight: 400;
        font-size: 20px;
        position: relative;
        color: #fff;
        &:after{
            content:"";
            width: 0;
            height: 2px;
            background-color: #fff;
            position: absolute;
            bottom: 0;
            left: 50%;
            transform: translateX(-50%);
            transition: all 0.3s;
          }
        &:hover:after{
            width: 100%;
          }
    }   

    .IntroLogo{
        transition: 0.5s;
        position: fixed;
        top: 0;
        left: 0;
        box-sizing: border-box;
        width: 100%;
        height: 100%;
        animation: opening 1.5s 0s linear;
        @include flexCenter;
        align-items: center;
        flex-direction: column;
        .logo{
            user-select: none;
            animation: bounce 2s ease;
            cursor: pointer;
            transition: 1s;
            box-sizing: border-box;
            width: 16vw;
            height: 16vw;
            margin: 0 auto;
            @include flexCenter;
            align-items: center;
            &:hover{
                animation: bounce 2s ease infinite;
            }
        }
        .logoTxt{
            user-select: none;
            cursor: pointer;
            transition: 1s;
            box-sizing: border-box;
            width: 42vw;
            height: 5vw;
            @include flexCenter;
            align-items: center;
        }
    }

    .Slide{
        transition: 0.5s;
        transition: 0.5s;
        position: fixed;
        top: 0;
        left: 0;
        box-sizing: border-box;
        width: 100%;
        height: 100%;
        .ant-carousel .slick-slide {
            @include flexCenter;
            align-items: center;
            text-align: center;
            height: 90vh;
            background: #244949;
            overflow: hidden;
            @include flexCenter;
            align-items: center;
        }
        .slideTextBox{
            padding-top: 20px;
            margin-left: 20px;
            margin-right: 20px;
        }
        .introSlide{
            box-sizing: border-box;
            margin: 0 auto;
            width: 40vw;
            height: 30vw;
            max-width: 600px;
            max-height: 360px;
            &.logoSlide{
                width: 38vw;
                height: 22vw;
                max-width: 520px;
                max-height: 290px;
                padding-bottom: 10px;
            }
            &.introSlide0{
                width: 30vw;
                max-width: 360px;
            }
            &.introSlide2{
                height: 24vw;
                max-height: 320px;
            }
        }
        .iconLeft{
            position: absolute;
            left: 10vw;
            top: 50%;
            transform: translateY(-50%);
            font-size: 30px;
            color: #fff;
            z-index: 300;
        }
        .iconRight{
            position: absolute;
            right: 10vw;
            top: 50%;
            transform: translateY(-50%);
            font-size: 30px;
            color: #fff;
            z-index: 300;
        }
        .btnSkip{
            user-select: none;
            position: absolute;
            left: 50%;
            bottom: 12vh;
            transform: translate(-50%,-50%);
            cursor: pointer;
            transition: 1s;
            font-weight: 400;
            font-size: 20px;
            color: #fff;
            &:after{
                content:"";
                width: 0;
                height: 2px;
                background-color: #fff;
                position: absolute;
                bottom: 0;
                left: 50%;
                transform: translateX(-50%);
                transition: all 0.3s;
            }
            &:hover:after{
                width: 100%;
            }
        }
    }

    

    
    
}