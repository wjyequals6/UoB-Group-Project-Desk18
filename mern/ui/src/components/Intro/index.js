import React, { useState, useEffect } from "react";
import "./Intro.scss";
import logo_removebg from "../../img/logo_removebg.png";
import logoTxt_removebg from "../../img/logoTxt_removebg.png";
import logoSlide from "../../img/logoSlide.png";
import introSlide0 from "../../img/introSlide0.png";
import introSlide1 from "../../img/introSlide1.png";
import introSlide2 from "../../img/introSlide2.gif";
import introSlide3 from "../../img/introSlide3.gif";


import { Carousel } from 'antd';
import { LeftOutlined, RightOutlined } from '@ant-design/icons';
import 'antd/dist/antd.css';

const Intro = (props) => {
    const [showIntro, setShowIntro] = useState(true);
    const [showLogo, setShowLogo] = useState(true);
    const [showSlide, setShowSlide] = useState(false);

    

    const carouselRef = React.createRef();

    const handleNext = () => carouselRef.current.next();

    const handlePrev = () => carouselRef.current.prev();

    const {  } = props;
    return (
       <div className="Intro" data-testid="Intro" style={showIntro ? {visibility: "visible", opacity: 1 } : {visibility: "hidden", opacity: 0}}  >
           <div className="IntroLogo" style={showLogo ? {visibility: "visible", opacity: 1 } : {visibility: "hidden", opacity: 0}}>
                <img src={logo_removebg} data-testid="logo" className="logo" onClick={() => {setShowLogo(false);setShowSlide(true); }} />
                <img src={logoTxt_removebg} data-testid="logoTxt" className="logoTxt" onClick={() => {setShowLogo(false);setShowSlide(true); }} />
                <div><div className="btnText startText" data-testid="btnStart" onClick={() => {setShowIntro(false); setShowLogo(false); setShowSlide(false);}}>START</div></div>
                <div><div className="btnText tutorialText" data-testid="btnTutorial" onClick={() => {setShowLogo(false);setShowSlide(true); }}>TUTORIAL</div></div>
           </div>
           <div className="Slide" style={showSlide ? {visibility: "visible", opacity: 1 } : {visibility: "hidden", opacity: 0}}>
                <LeftOutlined className="iconLeft" onClick={handlePrev}  />     
                <Carousel ref={carouselRef}>
                    <div>
                        <img className="introSlide logoSlide" src={logoSlide}></img>
                        <div className="slideTextBox"><h1>COVIDSURVIVAL IS A PLATFORM FOR EVERYONE TO LEARNING THE INFROMATION OF COVID-19 AND SHARE YOUR THOUGHTS</h1></div>
                    </div>
                    <div>
                        <img className="introSlide introSlide0" src={introSlide0}></img>
                        <div className="slideTextBox"><h1>THE GLOBAL STATICS ARE SHOWN ON THE TOP LEFT OF THE PAGE</h1></div>
                    </div>
                    <div>
                        <img className="introSlide introSlide1" src={introSlide1}></img>
                        <div className="slideTextBox"><h1>ONE CLICK ON MAP FOR REAL-TIME OFFICIAL STATISTICS OF COVID-19</h1></div>
                    </div>
                    <div>
                        <img className="introSlide introSlide2" src={introSlide2}></img>
                        <div className="slideTextBox"><h1>FURTHER DETAILS OF TRAVEL POLICY AND NEWS ARE ON THE BOTTOM-LEFT OF THE PAGE</h1></div>
                    </div>
                    <div>
                        <img className="introSlide introSlide3" src={introSlide3}></img>
                        <div className="slideTextBox"><h1>AFTER SIGN-IN, YOU CAN ACCESS THE FORUM TO SHARE THE TIPS TO SURVIVE THE COVID</h1></div>
                    </div>
               </Carousel>
               <RightOutlined className="iconRight" onClick={handleNext}/>
               <div><div className="btnSkip" data-testid="btnSkip" onClick={() => {setShowIntro(false); setShowLogo(false); setShowSlide(false);}}>SKIP AND START</div></div>
           </div>
       </div>
    );
};

export default Intro;