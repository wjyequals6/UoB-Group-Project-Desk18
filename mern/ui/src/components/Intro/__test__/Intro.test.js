import React from 'react';
import ReactDOM from 'react-dom';
import Intro from './../index';
import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import userEvent from '@testing-library/user-event';


afterEach(cleanup); 

it("renders without crashing", ()=>{
    const div = document.createElement("div");
    ReactDOM.render(<Intro/>, div);
});

it("check all component render successfully", ()=>{
    const {getByTestId} = render(<Intro/>);
    expect(getByTestId('Intro')).toBeTruthy();
});

it("render text correctly", () => {
    const {getByTestId} = render(<Intro/>);
    expect(getByTestId('btnStart')).toHaveTextContent("START");
    expect(getByTestId('btnTutorial')).toHaveTextContent("TUTORIAL");
    expect(getByTestId('btnSkip')).toHaveTextContent("SKIP AND START");
});

it("match snapshot", () => {
    const tree = renderer.create(<Intro/>).toJSON();
    expect(tree).toMatchSnapshot();
});

