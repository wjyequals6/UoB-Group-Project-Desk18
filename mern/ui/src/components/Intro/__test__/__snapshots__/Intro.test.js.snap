// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="Intro"
  data-testid="Intro"
  style={
    Object {
      "opacity": 1,
      "visibility": "visible",
    }
  }
>
  <div
    className="IntroLogo"
    style={
      Object {
        "opacity": 1,
        "visibility": "visible",
      }
    }
  >
    <img
      className="logo"
      data-testid="logo"
      onClick={[Function]}
      src="logo_removebg.png"
    />
    <img
      className="logoTxt"
      data-testid="logoTxt"
      onClick={[Function]}
      src="logoTxt_removebg.png"
    />
    <div>
      <div
        className="btnText startText"
        data-testid="btnStart"
        onClick={[Function]}
      >
        START
      </div>
    </div>
    <div>
      <div
        className="btnText tutorialText"
        data-testid="btnTutorial"
        onClick={[Function]}
      >
        TUTORIAL
      </div>
    </div>
  </div>
  <div
    className="Slide"
    style={
      Object {
        "opacity": 0,
        "visibility": "hidden",
      }
    }
  >
    <span
      aria-label="left"
      className="anticon anticon-left iconLeft"
      onClick={[Function]}
      role="img"
      tabIndex={-1}
    >
      <svg
        aria-hidden="true"
        data-icon="left"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
        />
      </svg>
    </span>
    <div
      className="ant-carousel"
    >
      <div
        className="slick-slider slick-initialized"
        dir="ltr"
      >
        <div
          className="slick-list"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={null}
          onMouseMove={null}
          onMouseUp={[Function]}
          onTouchCancel={null}
          onTouchEnd={[Function]}
          onTouchMove={null}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="slick-track"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "WebkitTransform": "translate3d(0px, 0px, 0px)",
                "WebkitTransition": "",
                "msTransform": "translateX(0px)",
                "opacity": 1,
                "transform": "translate3d(0px, 0px, 0px)",
                "transition": "",
              }
            }
          >
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={-1}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    className="introSlide introSlide3"
                    src="introSlide3.gif"
                  />
                  <div
                    className="slideTextBox"
                  >
                    <h1>
                      AFTER SIGN-IN, YOU CAN ACCESS THE FORUM TO SHARE THE TIPS TO SURVIVE THE COVID
                    </h1>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={false}
              className="slick-slide slick-active slick-current"
              data-index={0}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    className="introSlide logoSlide"
                    src="logoSlide.png"
                  />
                  <div
                    className="slideTextBox"
                  >
                    <h1>
                      COVIDSURVIVAL IS A PLATFORM FOR EVERYONE TO LEARNING THE INFROMATION OF COVID-19 AND SHARE YOUR THOUGHTS
                    </h1>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide"
              data-index={1}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    className="introSlide introSlide0"
                    src="introSlide0.png"
                  />
                  <div
                    className="slideTextBox"
                  >
                    <h1>
                      THE GLOBAL STATICS ARE SHOWN ON THE TOP LEFT OF THE PAGE
                    </h1>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide"
              data-index={2}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    className="introSlide introSlide1"
                    src="introSlide1.png"
                  />
                  <div
                    className="slideTextBox"
                  >
                    <h1>
                      ONE CLICK ON MAP FOR REAL-TIME OFFICIAL STATISTICS OF COVID-19
                    </h1>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide"
              data-index={3}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    className="introSlide introSlide2"
                    src="introSlide2.gif"
                  />
                  <div
                    className="slideTextBox"
                  >
                    <h1>
                      FURTHER DETAILS OF TRAVEL POLICY AND NEWS ARE ON THE BOTTOM-LEFT OF THE PAGE
                    </h1>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide"
              data-index={4}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    className="introSlide introSlide3"
                    src="introSlide3.gif"
                  />
                  <div
                    className="slideTextBox"
                  >
                    <h1>
                      AFTER SIGN-IN, YOU CAN ACCESS THE FORUM TO SHARE THE TIPS TO SURVIVE THE COVID
                    </h1>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={5}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    className="introSlide logoSlide"
                    src="logoSlide.png"
                  />
                  <div
                    className="slideTextBox"
                  >
                    <h1>
                      COVIDSURVIVAL IS A PLATFORM FOR EVERYONE TO LEARNING THE INFROMATION OF COVID-19 AND SHARE YOUR THOUGHTS
                    </h1>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={6}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    className="introSlide introSlide0"
                    src="introSlide0.png"
                  />
                  <div
                    className="slideTextBox"
                  >
                    <h1>
                      THE GLOBAL STATICS ARE SHOWN ON THE TOP LEFT OF THE PAGE
                    </h1>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={7}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    className="introSlide introSlide1"
                    src="introSlide1.png"
                  />
                  <div
                    className="slideTextBox"
                  >
                    <h1>
                      ONE CLICK ON MAP FOR REAL-TIME OFFICIAL STATISTICS OF COVID-19
                    </h1>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={8}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    className="introSlide introSlide2"
                    src="introSlide2.gif"
                  />
                  <div
                    className="slideTextBox"
                  >
                    <h1>
                      FURTHER DETAILS OF TRAVEL POLICY AND NEWS ARE ON THE BOTTOM-LEFT OF THE PAGE
                    </h1>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={9}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    className="introSlide introSlide3"
                    src="introSlide3.gif"
                  />
                  <div
                    className="slideTextBox"
                  >
                    <h1>
                      AFTER SIGN-IN, YOU CAN ACCESS THE FORUM TO SHARE THE TIPS TO SURVIVE THE COVID
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <ul
          className="slick-dots slick-dots-bottom"
          onMouseEnter={null}
          onMouseLeave={null}
          onMouseOver={null}
          style={
            Object {
              "display": "block",
            }
          }
        >
          <li
            className="slick-active"
          >
            <button
              onClick={[Function]}
            >
              1
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              2
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              3
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              4
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              5
            </button>
          </li>
        </ul>
      </div>
    </div>
    <span
      aria-label="right"
      className="anticon anticon-right iconRight"
      onClick={[Function]}
      role="img"
      tabIndex={-1}
    >
      <svg
        aria-hidden="true"
        data-icon="right"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
        />
      </svg>
    </span>
    <div>
      <div
        className="btnSkip"
        data-testid="btnSkip"
        onClick={[Function]}
      >
        SKIP AND START
      </div>
    </div>
  </div>
</div>
`;
